containers/payments/subscription/SubscriptionSection.js

```diff
@@ -83,37 +77,66 @@ const SubscriptionSection = ({ permission }) => {
         );
     }
 
+    const {
+        UsedDomains,
+        MaxDomains,
+        UsedSpace,
+        MaxSpace,
+        UsedAddresses,
+        MaxAddresses,
+        UsedMembers,
+        MaxMembers,
+        MaxVPN
+    } = organization || {};
+
     const { mailPlan, vpnPlan } = formatPlans(Plans);
     const { Name: mailPlanName } = mailPlan || {};
     const { Name: vpnPlanName } = vpnPlan || {};
-    const canRemoveCoupon = CouponCode && ![BUNDLE, BLACK_FRIDAY.COUPON_CODE].includes(CouponCode);
-    const i18n = getCyclesi18n();
-
-    /**
-     * Open subscription modal with criteria
-     * @param {String} action remove-coupon, yearly, upgrade, update
-     */
-    const handleModal = (action = '') => () => {
-        const coupon = action === 'remove-coupon' ? '' : CouponCode ? CouponCode : undefined; // CouponCode can equals null
-        const cycle = action === 'yearly' ? YEARLY : Cycle;
-        const plansMap = action === 'upgrade' ? { plus: 1, vpnplus: 1 } : toPlanNames(subscription.Plans);
 
+    const handleModal = () => {
         createModal(
             <SubscriptionModal
                 subscription={subscription}
-                plansMap={plansMap}
-                coupon={coupon}
+                plansMap={toPlanNames(subscription.Plans)}
+                coupon={CouponCode || undefined} // CouponCode can equals null
                 currency={Currency}
-                cycle={cycle}
+                cycle={Cycle}
             />
         );
     };
 
+    const mailAddons = [
+        hasPaidMail && { label: c('Label').t`Users`, used: UsedMembers, max: MaxMembers },
+        hasPaidMail && { label: c('Label').t`Email addresses`, used: UsedAddresses, max: MaxAddresses },
+        hasPaidMail && {
+            label: c('Label').t`Storage capacity`,
+            used: UsedSpace,
+            max: MaxSpace,
+            humanSize,
+            format: (v) => humanSize(v, 'GB')
+        },
+        hasPaidMail && { label: c('Label').t`Custom domains`, used: UsedDomains, max: MaxDomains },
+        mailPlanName === 'visionary' && { label: c('Label').t`VPN connections`, max: MaxVPN }
+    ].filter(Boolean);
+
+    const vpnAddons = [hasPaidVpn && { label: c('Label').t`VPN connections`, max: MaxVPN }].filter(Boolean);
```

Comment:
Is it possible that the `VPN connections` can be displayed twice (if both `mailPlanName === 'visionary'` and `hasPaidVpn = true`)?

Reply:
Nope:
```jsx
{mailPlanName === 'visionary' ? null : (
                    <div className="pt1 pl1 pr1">
                        <div className="flex-autogrid onmobile-flex-column w100 mb1">
                            <div className="flex-autogrid-item">ProtonVPN plan</div>
                            <div className="flex-autogrid-item">
                                <strong>{hasPaidVpn ? PLAN_NAMES[vpnPlanName] : c('Plan').t`Free`}</strong>
                            </div>
                            <div className="flex-autogrid-item" />
                        </div>
                        {vpnAddons.map((props, index) => (
                            <AddonRow key={index} {...props} />
                        ))}
                    </div>
                )}
```